packages/superset-ui-chart-controls/src/operators/timeComparePivotOperator.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable camelcase */
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitationsxw
+ * under the License.
+ */
+import { ComparisionType, PostProcessingPivot, NumpyFunction } from '@superset-ui/core';
+import { getMetricOffsetsMap, isValidTimeCompare, TIME_COMPARISION } from './utils';
+import { PostProcessingFactory } from './types';
+
+export const timeComparePivotOperator: PostProcessingFactory<PostProcessingPivot | undefined> = (
+  formData,
+  queryObject,
+) => {
+  const comparisonType = formData.comparison_type;
+  const metricOffsetMap = getMetricOffsetsMap(formData, queryObject);
+
+  if (isValidTimeCompare(formData, queryObject)) {
+    const valuesAgg = Object.fromEntries(
+      Array.from(metricOffsetMap.values())
+        .concat(Array.from(metricOffsetMap.keys()))
```

Comment:
Can use array spread here, I think
```suggestion
      [...metricOffsetMap.values(), ...metricOffsetMap.keys()]
```